### DISCUSSION TITLE
How to set the Background and Foreground when DropDownButton is pressed?

### DISCUSSION THREAD

**Discussion by JackDjTom** [MahApps/MahApps.Metro#4063] [Q&A]
Hi, 
I can't figure out how to set the `Background` and the `Foreground` of the `DropDownButton` when it's pressed... can someone help me?

I have:
```
<Style x:Key="{x:Type mah:DropDownButton}"
         BasedOn="{StaticResource {x:Type mah:DropDownButton}}"
         TargetType="{x:Type mah:DropDownButton}">
      <Setter Property="ButtonStyle">
        <Setter.Value>
          <Style BasedOn="{StaticResource MahApps.Styles.Button.Flat}" TargetType="{x:Type ButtonBase}">
            <Style.Triggers>
              <Trigger Property="IsPressed" Value="True">
                <Setter Property="Background" Value="{DynamicResource MahApps.Brushes.Button.Flat.Background.Pressed}" />
                <Setter Property="Foreground" Value="{DynamicResource MahApps.Brushes.Button.Flat.Foreground.Pressed}" />
              </Trigger>
            </Style.Triggers>
          </Style>
        </Setter.Value>
      </Setter>
</Style>
```

But that does nothing...

And having the Triggers directly on the `DropDownButton`-`Style `doesn't work since the `DropDownButton` has no `IsPressed` Property.

I know that i can set the `ArrowPressedBrush` but i want to set the `Background` of the `Button` and the `Foreground` of the `Content` (Text) aswell...

**Comment by timunie** [comment]
Hi @JackDjTom 

I had the same need in the past too. And to be honest I ended up copying the `ControlTemplate` and set the needed `Triggers` there. 

https://github.com/MahApps/MahApps.Metro/blob/29a0ac40a9b99e192aeff63a6def6580c0203076/src/MahApps.Metro/Themes/DropDownButton.xaml#L37-L151

Maybe there is a better solution for it, but that is how I did it. 

Happy coding
Tim
